kotlin-native/runtime/src/main/cpp/ObjCExportCoroutines.mm

```diff
@@ -36,8 +36,8 @@
 extern "C" OBJ_GETTER(Kotlin_ObjCExport_createContinuationArgumentImpl,
     KRef completionHolder, const TypeInfo** exceptionTypes);
 
-extern "C" OBJ_GETTER(Kotlin_ObjCExport_createContinuationArgument, id completion, const TypeInfo** exceptionTypes) {
-  if (pthread_main_np() != 1) {
+extern "C" OBJ_GETTER(Kotlin_ObjCExport_createContinuationArgument, KBoolean allowAnyThread, id completion, const TypeInfo** exceptionTypes) {
```

Comment:
There is a significantly simpler way to pass the value from compiler arguments to runtime.
See e.g. https://github.com/JetBrains/kotlin/blob/19658b3a4706284eeb21480f53fe2740ba60ce88/kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt#L2779 and https://github.com/JetBrains/kotlin/blob/3798920183e9180795726ac89d18c4355dac4520/kotlin-native/runtime/src/main/cpp/CompilerConstants.cpp#L17

Please use this approach.

Reply:
done